joblib/parallel.py

```diff
@@ -692,65 +696,65 @@ def print_progress(self):
                          short_format_time(remaining_time),
                          ))
 
+    def _handle_error(self, exception):
+
+        print(exception)
+        # Stop dispatching any new job in the async callback thread
+        self._aborting = True
+
+        # If the backend allows it, cancel or kill remaining running
+        # tasks without waiting for the results as we will raise
+        # the exception we got back to the caller instead of returning
+        # any result.
+        backend = self._backend
+        if (backend is not None and
+                hasattr(backend, 'abort_everything')):
+            # If the backend is managed externally we need to make sure
+            # to leave it in a working state to allow for future jobs
+            # scheduling.
+            ensure_ready = self._managed_backend
+            backend.abort_everything(ensure_ready=ensure_ready)
+
+        if not isinstance(exception, TransportableException):
+            raise
+        else:
+            # Capture exception to add information on the local
+            # stack in addition to the distant stack
+            this_report = format_outer_frames(context=10,
+                                              stack_start=1)
+            report = """Multiprocessing exception:
+%s
+---------------------------------------------------------------------------
+Sub-process traceback:
+---------------------------------------------------------------------------
+%s""" % (this_report, exception.message)
+            # Convert this to a JoblibException
+            exception_type = _mk_exception(exception.etype)[0]
+            exception = exception_type(report)
+
+            raise exception
+
     def retrieve(self):
-        self._output = list()
         while self._iterating or len(self._jobs) > 0:
             if len(self._jobs) == 0:
                 # Wait for an async callback to dispatch new jobs
                 time.sleep(0.01)
                 continue
-            # We need to be careful: the job list can be filling up as
-            # we empty it and Python list are not thread-safe by default hence
-            # the use of the lock
+
+            # We need to be careful: the buffer list can be filling up as
+            # we empty it and Python list are not thread-safe by default
+            # hence the use of the lock
```

Comment:
The change above is a cosmetic one that, I believe, has nothing to do with the functionality added. It is wise to refrain from such a change when adding new functionality, as it make it harder to review.

Pure cosmetic PRs can be useful for such purpose.